service/resource/configmap/current.go

```diff
@@ -0,0 +1,34 @@
+package configmap
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/microerror"
+)
+
+// GetCurrentState returns the current state of the prometheus config configmap.
+// If the configmap exists, it is returned.
+// If the configmap does not exist, nil is returned.
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	r.logger.Log("debug", fmt.Sprintf("fetching configmap: %v/%v", r.configMapNamespace, r.configMapName))
+
+	configMap, err := r.k8sClient.CoreV1().ConfigMaps(r.configMapNamespace).Get(
+		r.configMapName, metav1.GetOptions{},
+	)
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
I still prefer the way we do error handling in all other microservices. It is flatter and less complex because you have only one level of indentation and you can read the conditions one after another. 

```
if errors.IsNotFound(err) {
  // 
} else if err != nil {
  //
}
```